packages/flutter/test/widgets/editable_text_test.dart

```diff
@@ -12544,6 +12544,150 @@ void main() {
+                controller: TextEditingController(text: 'A'),
+                focusNode: FocusNode(),
+                style: const TextStyle(),
+                cursorColor: Colors.blue,
+                backgroundCursorColor: Colors.grey,
+                cursorOpacityAnimates: true,
+                autofillHints: null,
+                spellCheckConfiguration:
+                  SpellCheckConfiguration(
+                    misspelledTextStyle: TextField.materialMisspelledTextStyle,
+                ),
+              ),
+            ),
+          );
+
+          final EditableTextState state =
+            tester.state<EditableTextState>(find.byType(EditableText));
+          expect(state.spellCheckEnabled, isTrue);
+          expect(
+            state.spellCheckConfiguration!.spellCheckService.runtimeType,
+            equals(DefaultSpellCheckService)
+          );
+          expect(
+            state.spellCheckConfiguration!.spellCheckSuggestionsHandler.runtimeType,
+            equals(DefaultSpellCheckSuggestionsHandler)
+          );
+        });
+
+    testWidgets(
+        'Spell check configured properly with specified spell check service and handler',
+            (WidgetTester tester) async {
+          final FakeSpellCheckService fakeSpellCheckService = FakeSpellCheckService();
+          final FakeSpellCheckSuggestionsHandler fakeSpellCheckSuggestionsHandler =
+          FakeSpellCheckSuggestionsHandler();
+
+          await tester.pumpWidget(
+            MaterialApp(
+              home: EditableText(
+                controller: TextEditingController(text: 'A'),
+                focusNode: FocusNode(),
+                style: const TextStyle(),
+                cursorColor: Colors.blue,
+                backgroundCursorColor: Colors.grey,
+                cursorOpacityAnimates: true,
+                autofillHints: null,
+                spellCheckConfiguration:
+                  SpellCheckConfiguration(
+                    spellCheckService: fakeSpellCheckService,
+                    spellCheckSuggestionsHandler: fakeSpellCheckSuggestionsHandler,
+                    misspelledTextStyle: TextField.materialMisspelledTextStyle,
+                ),
+              ),
+            ),
+          );
+
+          final EditableTextState state =
+            tester.state<EditableTextState>(find.byType(EditableText));
+          expect(
+            state.spellCheckConfiguration!.spellCheckService.runtimeType,
+            equals(FakeSpellCheckService)
+          );
+          expect(
+            state.spellCheckConfiguration!.spellCheckSuggestionsHandler.runtimeType,
+            equals(FakeSpellCheckSuggestionsHandler)
+          );
+        });
+
+    testWidgets(
+        'Error thrown when spell check enabled but no default spell check service available',
+            (WidgetTester tester) async {
+          tester.binding.platformDispatcher.nativeSpellCheckServiceDefinedTestValue =
+            false;
```

Comment:
I don't think this will impact anything else since spell check is disabled by default, but I will add it in preparation for it to be enabled by default.